### ISSUE TITLE
why we not support jdk 8?

### ISSUE THREAD

**Issue by iamcyw** [aileftech/snap-admin#33]
Not all production projects are new versions, most of us are still at jdk8.
please think about this

**Comment by aileftech** [comment]
You are right. I will make it so that the next release supports JDK >= 8. I don't wanna mess this up though, so I need to check carefully what needs to change.

For sure the `java.version = 17` in the `pom.xml` file. I guess I need to use JDK 8 to compile the project as well otherwise it will not run when installed with Maven?

**Comment by aileftech** [comment]
After some testing I realized I used more Java > 8 features than I expected. I still plan to do this but it will require a bit more time than expected.

**Comment by bobgarner** [comment]
I was able to bump my java version to 17 but now I am getting a runtime exception at startup:

```
Failed to introspect Class [tech.ailef.snapadmin.external.controller.SnapAdminController] from ClassLoader [jdk.internal.loader.ClassLoaders$AppClassLoader@531d72ca]
```

There is a bunch of `org.springframework.beans...` in the stack trace, do you have a minimum version for Spring Boot?

**Comment by aileftech** [comment]
I would need to test it, but the lowest I ever tried is 3.1.3. What version are you using?

**Comment by bobgarner** [comment]
Ah ok, I am at 2.5.5. That is way to low, I've been meaning to bring it up to date so I will do that.

**Comment by DaurenF** [comment]
HI! Doesn't it support old version of hibernate (5.x.x) versions?
I get this error after adding it to gradle : 

    // https://mvnrepository.com/artifact/tech.ailef/snap-admin
    implementation 'tech.ailef:snap-admin:0.2.0'
    
import javax.persistence.Entity;
error: cannot find symbol

**Comment by aileftech** [comment]
Hi, I have not tested with older Hibernate versions but what I can grasp from the error is that it might be related to the `javax.persistence` to `jakarta.persistence` renamed package. At the moment I don't have a solution for this though, I guess this would require shipping two different versions of the library? 

I'd really like to avoid this so If somebody with more experience is aware of a different method to ship just one version that supports both (assuming it's possible) it'd be awesome.

**Comment by ochompsky** [comment]
> Not all production projects are new versions, most of us are still at jdk8. please think about this

who is 'most'?  why ask the developer to spend his time making this amazing library work because you are too lazy to upgrade?
